dboperations/dboperations.go

```diff
@@ -0,0 +1,177 @@
+	if err != nil {
+		log.Printf("error in getting the Database value: %v", err)
+		return nil, err
+	}
+	var response models.ProductDetails
+	versionsArray := []string{}
+	for _, i := range res.Items {
+		err = dynamodbattribute.UnmarshalMap(i, &response)
+		versionsArray = append(versionsArray, response.Version)
+		if err != nil {
+			log.Printf("Got error unmarshalling: %s", err)
+			return nil, err
+		}
+
+	}
+	return versionsArray, nil
+}
+
+func (dbo *DbOperationsService) GetMetaData(partitionValue string, sortValue string, platform string, platformVersion string, architecture string) (*models.MetaData, error) {
+	input := &dynamodb.GetItemInput{
+		TableName: aws.String(dbo.productTableName),
+		Key: map[string]*dynamodb.AttributeValue{
+			PRODUCT_PARTITION_KEY: {S: aws.String(partitionValue)},
+			PRODUCT_SORT_KEY:      {S: aws.String(sortValue)},
+		},
+	}
+	res, err := dbo.db.GetItem(input)
+	if err != nil {
+		return nil, err
+	}
+	var productDetails models.ProductDetails
+	if err := dynamodbattribute.UnmarshalMap(res.Item, &productDetails); err != nil {
+		return nil, err
+	}
+	MetaData := productDetails.MetaData
+	var response models.MetaData
+	for _, j := range MetaData {
+		if j.Architecture == architecture && j.Platform == platform && j.Platform_Version == platformVersion {
+			response.Architecture = architecture
+			response.Platform = platform
+			response.Platform_Version = platformVersion
+			response.SHA1 = j.SHA1
+			response.SHA256 = j.SHA256
+		}
+	}
+	return &response, nil
+}
+
+func (dbo *DbOperationsService) GetVersionLatest(partitionValue string) (string, error) {
+	versions, err := dbo.GetVersionAll(partitionValue)
+	if err != nil {
+		log.Printf("Error in getting versions list: %v", err)
+		return "", err
+	}
+	sort.Sort(sort.Reverse(sort.StringSlice(versions)))
+	sortValue := versions[0]
+	latestVersionDetails, err := dbo.GetPackages(partitionValue, sortValue)
+	if err != nil {
+		log.Printf("Error in fetching the latest version: %v", err)
+		return "", err
+	}
+	return latestVersionDetails.Version, nil
+}
+
+func (dbo *DbOperationsService) GetRelatedProducts(partitionValue string) (*models.RelatedProducts, error) {
+	res, err := dbo.fetchDataValues(partitionValue, dbo.skuTableName)
+	if err != nil {
+		log.Printf("error in fetching the database values: %v", err)
+		return nil, err
+	}
+	var sku models.RelatedProducts
+	var responseArray []string
+	for _, i := range res.Items {
+		err = dynamodbattribute.UnmarshalMap(i, &sku)
+		responseArray = append(responseArray, sku.Products...)
+		if err != nil {
+			log.Printf("Got error unmarshalling: %s", err)
+			return nil, err
+		}
+	}
+	sku.Sku = partitionValue
+	sku.Products = responseArray
+
+	return &sku, nil
+}
+
+func (dbo *DbOperationsService) fetchDataValues(partitionValue string, tableName string) (*dynamodb.ScanOutput, error) {
+	filt := expression.Name(SKU_PARTITION_KEY).Equal(expression.Value(partitionValue))
```

Comment:
take PARTITION_KEY as  input as this function is called from two function and each time partition key value is different. GetVersionAll and GetRelatedProducts functions
Also variable should be `filter` instead for `filt`.

Reply:
Done